### DISCUSSION TITLE
the Help message is not really helpful

### DISCUSSION THREAD

**Discussion by crypt47** [ihhub/fheroes2#5646] [General]
Hello, I'm running the program for the first time and it doesn't look very convinient. (Sorry, don't know why code blocks here smashes formating)

`crypt@witch ~ $ fheroes2 --help
fheroes2 engine, version: 0.9.17
fheroes2: illegal option -- -
Usage: fheroes2 [OPTIONS]
  -h		print this help message and exit
`
ok

`
crypt@witch ~ $ fheroes2 -h
fheroes2 engine, version: 0.9.17
Usage: fheroes2 [OPTIONS]
  -h		print this help message and exit
crypt@witch ~ $ `

Imho, the help message should display more then this. OPTIONS description at least. Let's try more.

`
crypt@witch ~ $ man fheroes2
No manual entry for fheroes2
`
still failure.:(( wtf?

`
crypt@witch ~ $ uname -a
FreeBSD witch.inter 12.3-RELEASE-p5 FreeBSD 12.3-RELEASE-p5 GENERIC  amd64
`
Ok, run it as it is. It says something mysterious about missing some data in some place. Wtf?! Just write in simple:

`
Couldn't find data directory X.  Please see http://url, man whatever i can access without further searching. Don't say look at README file, because it's location is unknown.
`
Is that so hard?

**Comment by ihhub** [comment]
Hi @crypt47 ,

Please be more respectful and write your messages in a more polite rather than demanding manner. This is an open source project based on voluntary efforts, not a paid commercial product.

These options are obsolete and the code related to them was developed before us. I think we added man page recently and we could update the output of the console. We will rework this in the future.

**Reply by crypt47** [reply]
ihhub, you are right, that's insulting and that's exactly how I felt. I know the project is open source, but it's also advertized to end users as a ready to try. Not alfa or beta. And it insulting to the end user when the project doesn't provide essentials. When I have users in mind, I do think about help messages and problem reporting. Secondly I'm aware it's an opensource project and that's why I'm ready to report problems. As for example I've written yesterday the FreeBSD port was missing dependances. Thank you.

**Reply by ihhub** [reply]
We will update in-game help function. However, we aren't responsible for FreeBSD build. If you could point exactly what is missing and help us to solve this issue it would be great!

As I wrote we recently added man page file to the project repository which locates at docs/fheroes2.6.in. Is it the file you're referring to?

**Reply by crypt47** [reply]
That's alright, I've notified the port mainteiner. Let me check the man.

**Reply by crypt47** [reply]
For some reason the man page isn't parsed right. I'm not sure about '@' symbols, may be some examples like "$HOME/..." would be helpful.
`
GAME DATA PATHS
       The following directories are searched as data root: #_SG

       @FHEROES2_DATA@
              The system-wide location that can be accessed by all users.
              #_EG
`
Secondly, does the game have any options? Is it configurable? Any path or debug keys? Thirdly, the INSTALL.md mentions XDG_DATA_HOME/fheroes2 (usually ~/.local/share/fheroes2) directory as supposed path to work, but download_demo_version.sh installs files in ~/.fheroes2. It's confusing. Also I see some file

`
.fheroes2/fheroes2.bin: data
`
that are preinstalled before games but not in the manual.

**Reply by crypt47** [reply]
Also the question about upper/lower is not explained in the man. If the orginal file on CD are in upper case, we probably need to convert the names?

**Reply by crypt47** [reply]
I understand only the first menu screen is scaled, there are no options to run the games scaled?

**Comment by oleg-derevenetz** [comment]
@ihhub

> These options are obsolete and the code related to them was developed before us.

They are not quite obsolete, there is another one useful option to specify debug flags, but it is available (and is shown in help message) only on debug builds.

@crypt47

> Sorry, don't know why code blocks here smashes formating

That's because you are [doing it wrong](https://github.github.com/gfm/#fenced-code-blocks).

> For some reason the man page isn't parsed right.

That's because it's not the man page itself, it's a template intended not for the end user, but for the port/package maintainer. Package maintainer is responsible for generating a proper man page using this template with all proper substitutions for the specific OS distribution.

> Secondly, does the game have any options? Is it configurable? Any path or debug keys?

It is configurable using GUI, config file and some environment variables described in the man page.

> Thirdly, the INSTALL.md mentions XDG_DATA_HOME/fheroes2 (usually ~/.local/share/fheroes2) directory as supposed path to work, but download_demo_version.sh installs files in ~/.fheroes2. It's confusing.

`INSTALL.md` is written for specific OSes and distributions. FreeBSD is not included in this list. `XDG_DATA_HOME` is used only on Linux.

> Also I see some file `.fheroes2/fheroes2.bin: data`

The `fheroes2.bin` is a binary config file.

> If the orginal file on CD are in upper case, we probably need to convert the names?

No, normally you should not need to do this. Simply copying (or using a helper script) should be enough. The game should deal with filename case sensitivity by itself.

> I understand only the first menu screen is scaled, there are no options to run the games scaled?

No. The displayed space of the adventure map will be bigger with larger resolutions.
